server/src/main/java/org/elasticsearch/plugins/RepositoryPlugin.java

```diff
@@ -37,16 +37,25 @@
      *
      * @param env The environment for the local node, which may be used for the local settings and path.repo
      *
-     * @param namedXContentRegistry the NamedXContentRegistry used for the {@link Repository.Factory}
+     * The key of the returned {@link Map} is the type name of the repository and
+     * the value is a factory to construct the {@link Repository} interface.
+     */
+    default Map<String, Repository.Factory> getRepositories(Environment env, NamedXContentRegistry namedXContentRegistry,
+                                                            ClusterService clusterService) {
+        return Collections.emptyMap();
+    }
+
+    /**
+     * Returns internal repository types added by this plugin. Internal repositories cannot be registered
+     * through the external API.
      *
      * @param threadPool the thread pool used to execute the operations in the repositories
      *
      * The key of the returned {@link Map} is the type name of the repository and
      * the value is a factory to construct the {@link Repository} interface.
      */
-    default Map<String, Repository.Factory> getRepositories(Environment env,
-                                                            NamedXContentRegistry namedXContentRegistry,
-                                                            ThreadPool threadPool) {
+    default Map<String, Repository.Factory> getInternalRepositories(Environment env, NamedXContentRegistry namedXContentRegistry,
```

Comment:
We could probably leave this out as no `RepositoryPlugin` implementation actually uses it